File: javascript/prime-factors/prime-factors.js
export const primeFactors = (number) => {

  const factors = []
  let current_factor = 2;

  while (number > 1) {

    if (number % current_factor === 0) {
      
      factors.push(current_factor);
      number /= current_factor;
    }

    else
      current_factor++;
  }

  return factors;
};

File: csharp/prime-factors/PrimeFactors.cs
using System.Collections.Generic;

public static class PrimeFactors
{
    public static long[] Factors(long number)
    {
        var factors = new List<long>();
        for (long factor = 2; factor <= number; factor++)
        {
            while (number % factor == 0)
            {
                factors.Add(factor);
                number /= factor;
            }
        }
        return factors.ToArray();
    }
}